### PR TITLE
Update import-assertions tests to new semantics

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
This PR is best reviewed commit-by-commit.

1. _Update import-assertions tests to new semantics_: the [import attributes](https://github.com/tc39/proposal-import-attributes) TC39 proposal, previously known as "import assertions", now throws on unrecognized keys.
2. _Copy import-assertions tests to import-attributes (`with`)_: the proposal now uses the `with` keyword instead of `assert`. `assert` is still an optional part of the proposal, but engines are discouraged from implementing it and it's just a syntactic alias for `with`. The `assert` tests could be eventually removed from WPT (it similar to how, for example, tests test `document.createElement` but not `document["createElement"]`), but for now I'm keeping them for engines that currently only support `assert`.
3. _Copy `{css,json}-module` to `{css,json}-module-assertions`_ is just copy/pasting in preparation of the next commit
4. _`assert` -> `with` in `{css,json}-module` folders_ changes the keyword from `assert` to `with` in all those tests. I chose to go with the `*-module`/`*-module-assertions` folder names, and not `*-module-attributes`/`*-module-assertions` or ``*-module-attributes`/`*-module`, because `with` should be considered the default.

Closes https://github.com/tc39/proposal-import-attributes/issues/146

### CODE REVIEW

**Review by domenic — APPROVED** [review]
This looks good generally. Possible bonus points, not necessary to address before merging:

- Add a README.md to the parent folder explaining the situation. E.g.: what the exact parallel directory structures are; why things are duplicated; that you should generally try to duplicate any future changes appropriately; and under what conditions we can remove the `assert`-based folders in the future.

- Move the `resources/` folder into the parent folder, probably with some renames, to avoid duplicating it.

- Assuming we're planning to rebase-and-merge, I don't understand why the copy-and-update process for import-assertions is one step (commit 2 in the sequence), whereas for CSS/JSON modules it's two steps (commit 3 and 4).

If you'd rather we just merge these as-is, say the word, and I'll do that. The only one that I'd really like to see addressed is the README, and that can be done in a followup.

**Comment by domenic** [comment]
Hmm it might also be worth checking the test result changes. E.g. the `relative-urls` test appears to have started failing in Chromium for css-modules? https://wpt.fyi/results/html/semantics/scripting-1/the-script-element?diff&filter=ADC&run_id=5109198436696064&run_id=5184245088124928 And for some reason there are more tests of the `with` variety than the `assert` variety?

**Comment by nicolo-ribaudo** [comment]
> the relative-urls test appears to have started failing in Chromium for css-modules? [wpt.fyi/results/html/semantics/scripting-1/the-script-element?diff&filter=ADC&run_id=5109198436696064&run_id=5184245088124928](https://wpt.fyi/results/html/semantics/scripting-1/the-script-element?diff&filter=ADC&run_id=5109198436696064&run_id=5184245088124928)

This actually caught some missing replacements in the tests to match the new folders structure. Fixes in [`dfbf6c5` (#41156)](https://github.com/web-platform-tests/wpt/pull/41156/commits/dfbf6c55439a034a2776223ba14748c49240e19c). 

> Move the resources/ folder into the parent folder, probably with some renames, to avoid duplicating it.

There are many files there that still use the `assert`/`with` syntax, so it would only be possible to hoist _part_ of it.

---

I'm waiting for CI to finish to check if there are other problems, then I'll squash/rebase and push as three commits:
1. Update tests to the new semantics
2. Rename the existing folders using `assert`
3. Add tests for `with`

**Comment by nicolo-ribaudo** [comment]
> And for some reason there are more tests of the with variety than the assert variety?

For some reasons, `import-assertions/invalid-type-assertion-error.html` and `import-assertions/empty-assertion-clause.html` are not picked up. However, `import-assertions/unsupported-assertion.html` is. I do not understand why 🤔

**Comment by domenic** [comment]
Can you explain the failure at https://wpt.fyi/results/html/semantics/scripting-1/the-script-element/css-module-assertions/import-css-module-basic.html?diff&filter=ADC&run_id=5154967017488384&run_id=5116567409393664 ?

**Comment by nicolo-ribaudo** [comment]
@domenic Fixed, I accidentally changed an URL in the wrong way in that test, preventing the iframe from loading.

File: html/semantics/scripting-1/the-script-element/README.md
Comment (domenic) [inline html/semantics/scripting-1/the-script-element/README.md:16]: This is not the correct criteria. The correct criteria is when we've determined that `assert` is not required for web compatibility, and then proceed to remove it from the spec.

In other words, if it's in the spec, it needs to be tested; if it's not in the spec, it doesn't.

Comment (nicolo-ribaudo) [reply html/semantics/scripting-1/the-script-element/README.md:16]: Even if it's just a syntactic alias for with? I assumed it was similar to how, for example, wpt doesn't need to test what `document["getElementById"]` and `document?.getElementById` do if it's already testing `document.getElementById`.

Comment (domenic) [reply html/semantics/scripting-1/the-script-element/README.md:16]: We need test coverage for both paths (like we do for multiple types of JS property access). Whether we need a full duplicate test suite is debatable, but it shouldn't be possible to get 100% without implementing `with`, while `with` is in the spec.

Comment (nicolo-ribaudo) [reply html/semantics/scripting-1/the-script-element/README.md:16]: No, but it should be possible to get 100% without implementing assert even if assert is in the spec, because it's marked as "optional, do not implement this unless you really need to" in the spec.

Anyway, this?
> The web compatibility of removing the `assert` keyword is being investigated. If it will be deemed feasible, it will be removed from the proposal and the `assert` tests can be deleted.

Comment (domenic) [reply html/semantics/scripting-1/the-script-element/README.md:16]: Sorry, I misspoke and meant to write `assert`. I'm not sure I agree with that working mode, but regardless, your new wording SGTM.

Comment (nicolo-ribaudo) [reply html/semantics/scripting-1/the-script-element/README.md]: Readme updated!

**Pull request merged**
